### DISCUSSION TITLE
翻译词汇讨论与投票

### DISCUSSION THREAD

**Discussion by greyli** [greyli/flask-docs-zh#12] [Vote]
有一些词汇我暂时不确定应该用哪一个中文词汇（比如 blueprint 译作 「蓝本」还是「蓝图」），所以放到这里投票和讨论决定。欢迎投票（用对应的表情符号）和讨论，如果你有更好的翻译词语选择，欢迎在对应的评论下回复指出。你也可以补充你遇到的不确定词汇。

**Comment by greyli** [comment]
**blueprint**

- 蓝本 👀
- 蓝图 🚀
- blueprint（不译）♥

**Reply by frostming** [reply]
蓝图

**Reply by HOLLYwyh** [reply]
蓝图！

**Reply by ghost** [reply]
蓝图

**Reply by codists** [reply]
蓝图

**Reply by wrywrywry0890** [reply]
蓝图

**Comment by greyli** [comment]
**application**

- 程序 👀
- 应用 🚀

同时对应「程序上下文」和「应用上下文」。

示例：
- Web application
- WSGI application
- Application context
- Just remember that your Flask application object is the actual WSGI application.

**Reply by frostming** [reply]
应用

**Reply by andyzhouty** [reply]
前者：应用；后者：程序上下文，我觉得这样比较通顺

**Reply by RRRoger** [reply]
应用

**Reply by HOLLYwyh** [reply]
emmmm我感觉Application这个词在不同语境下可以翻译成不同的内容吧。这里我感觉应用好一点。

**Reply by ghost** [reply]
应用

**Reply by codists** [reply]
应用

**Reply by wrywrywry0890** [reply]
应用

**Comment by greyli** [comment]
**context**

- 上下文 👀
- context（不译） 🚀

**Reply by frostming** [reply]
上下文

**Reply by RRRoger** [reply]
上下文

**Reply by HOLLYwyh** [reply]
上下文

**Reply by ghost** [reply]
上下文

**Reply by codists** [reply]
上下文

**Reply by wrywrywry0890** [reply]
上下文

**Comment by greyli** [comment]
**serve**

- 提供 👀
- serve（不译） 🚀

**Reply by frostming** [reply]
这个词灵活，提供点上下文？

**Reply by greyli** [reply]
> 这个词灵活，提供点上下文？

- Jinja is a template language that renders the pages your application serves.
- How do I serve the static files in deployment?

**Reply by greyli** [reply]
我考虑这个词还是按照上下文译作「服务」或「提供」比较好。

**Reply by frostming** [reply]
动词不做规定比较好，灵活翻译

可以专注于名词 弄个glossary译名对照表给译者参考

**Reply by greyli** [reply]
嗯，有时间弄。上面那两句你会怎么译？

**Reply by frostming** [reply]
Jinja是应用中渲染页面使用的模板语言
如何在部署阶段服务静态文件

不强求直译

**Reply by ghost** [reply]
这个词还是结合具体场景做翻译吧

**Reply by codists** [reply]
提供

**Reply by wrywrywry0890** [reply]
提供

**Comment by greyli** [comment]
**login**

- 登陆 👀
- 登录 🚀

**Reply by imoyao** [reply]
[登入界面中，「登录」和「登陆」哪个正确？ - 知乎](https://www.zhihu.com/question/19570377)

**Reply by frostming** [reply]
登录

**Reply by andyzhouty** [reply]
登录

**Reply by Tridagger** [reply]
登录

**Reply by ghost** [reply]
登录

**Reply by codists** [reply]
登录。
参考：《现代汉语词典（第 7 版）》“登录”词条第 2 个解释——指计算机用户输入用户名和密码，以取得计算机网络系统的认可。

**Reply by FarmerChillax** [reply]
登录

**Reply by wrywrywry0890** [reply]
登录

**Comment by greyli** [comment]
**production**

- 生产 👀
- 线上 🚀
- production（不译）♥

**Reply by frostming** [reply]
生产

**Reply by 180909** [reply]
这个得看受众面的，个人觉得`线上`这个词受众面更广一点。

**Reply by zhengxiaowai** [reply]
建议不翻译，没有歧义代表的意思都是差不多，更加符合使用环境

**Reply by RRRoger** [reply]
建议不翻译

**Reply by codists** [reply]
生产

**Reply by wrywrywry0890** [reply]
生产

**Comment by greyli** [comment]
**you**

- 你 👀
- 您 🚀

**Reply by 180909** [reply]
你

**Reply by Tridagger** [reply]
你

**Reply by codists** [reply]
你

**Reply by wrywrywry0890** [reply]
你

**Comment by greyli** [comment]
**Thread-Locals** in Flask

- 本地线程 👀
- 本地线程对象 🚀
- 线程局部 ♥
- 线程局部变量 🎉
- 线程局部存储 😕（来自 #14）

**Reply by greyli** [reply]
那 Thread-Locals Objects 就译作「线程局部对象」？

**Reply by codists** [reply]
线程局部对象。
解释：
1、Python 官方文档将 “Thread-Locals Objects ” 翻译为“线程本地”，见：https://docs.python.org/zh-cn/3.9/library/threading.html#thread-local-data。个人认为不好，“本地”往往是相对于“线上”而言，容易产生误解。
2、“Thread-Locals  Objects” 应该是 “Thread-Local Objects ”，local 在这里作形容词，想表达的意思是：The Objects are local to Thread。即这些对象对于线程而言是局部的，在线程间是不可共享的（不是全局的）。跟这个意思相近的是局部变量，所以翻译成线程局部对象更好。

**Reply by johnson329** [reply]
本地线程

**Reply by wrywrywry0890** [reply]
线程局部对象

**Comment by andyzhouty** [comment]
URL

- 网址 :eyes:
- URL（不译） :rocket: 
- 视情况交替使用 :heart:

**Reply by 180909** [reply]
网址

**Reply by c9n** [reply]
URL

**Reply by Tridagger** [reply]
URL

**Reply by ceaper** [reply]
URL

**Reply by RRRoger** [reply]
URL

**Reply by HOLLYwyh** [reply]
URL

**Reply by ghost** [reply]
不译

**Reply by codists** [reply]
URL

**Reply by wrywrywry0890** [reply]
URL

**Comment by greyli** [comment]
error handler

**Reply by frostming** [reply]
错误处理函数

**Reply by zhengxiaowai** [reply]
「错误处理」
@frostming 有时候并不是一个函数

**Reply by frostming** [reply]
明白，但是error handling占去 了「错误处理」的翻译，叫「错误处理器」又不利于初学者明白这是什么

**Reply by codists** [reply]
错误处理器。

**Reply by Insutanto** [reply]
错误处理器

**Reply by FarmerChillax** [reply]
错误控制器

**Reply by lrsgzs** [reply]
错误处理器

**Reply by wrywrywry0890** [reply]
错误处理器

**Comment by greyli** [comment]
before request function

**Reply by frostming** [reply]
请求前钩子（函数）

**Reply by FarmerChillax** [reply]
请求前过滤函数(过滤器)

**Reply by wrywrywry0890** [reply]
请求前函数

**Comment by greyli** [comment]
after request function

**Reply by frostming** [reply]
请求后钩子（函数）

**Reply by wrywrywry0890** [reply]
请求后函数

**Comment by greyli** [comment]
teardown function

**Reply by frostming** [reply]
清理函数

**Reply by codists** [reply]
清理函数

**Reply by wrywrywry0890** [reply]
清理函数

**Comment by greyli** [comment]
issue tracker

**Reply by greyli** [reply]
当前没有翻译

**Reply by frostming** [reply]
问题追踪（页面）

**Reply by greyli** [reply]
译作「问题追踪」还不错。不过我担心这样翻译会不会导致用户用 issue tracker 来提问？

**Reply by frostming** [reply]
或者是“事务追踪”

**Reply by wrywrywry0890** [reply]
问题追踪

**Comment by greyli** [comment]
PyPI releases

**Reply by greyli** [reply]
当前没有翻译

**Reply by zhengxiaowai** [reply]
PyPI 发布？

**Reply by greyli** [reply]
我考虑可以在「PyPI 发布」和「PyPI 发行」之间选一个。

**Reply by FarmerChillax** [reply]
> 我考虑可以在「PyPI 发布」和「PyPI 发行」之间选一个。

**打包发布**是否更贴切？

**Reply by wrywrywry0890** [reply]
PyPI发布

**Comment by guiqiqi** [comment]
**Debug**

- 调试 👀
- Debug（不译）🚀

**Reply by wrywrywry0890** [reply]
调试

**Comment by guiqiqi** [comment]
**Config**

- 配置 👀
- 设置🚀

**Reply by frostming** [reply]
Config配置
Settings设置

**Reply by wrywrywry0890** [reply]
配置

**Comment by rosekc** [comment]
worker

上下文参见 #65 

- 工作者 👀
- 不译 🚀
- 例程 #24 ♥

**Reply by johnson329** [reply]
不译

**Reply by wrywrywry0890** [reply]
例程

**Comment by ghost** [comment]
Lazy or Eager Loading

- 不译👀
- 懒惰加载或急切加载🚀
- 惰加载或急加载🎉

**Reply by rosekc** [reply]
即时加载？

**Reply by ghost** [reply]
延迟加载和即时加载会好些吗？

**Reply by rosekc** [reply]
懒惰加载似乎用的人更多？按照其他领域的资料与懒惰加载对应的词是预加载，但根据翻译的上下文好像并没有预加载的感觉，所以我倾向于即时加载。

**Reply by guiqiqi** [reply]
懒加载和预加载用的最多了 :)

**Reply by wrywrywry0890** [reply]
懒加载或急切加载

**Comment by yuxiaoy1** [comment]
hook point
- 钩点 👀 (hook 一般译为钩子)
- 挂接点 🚀 (参考这个 [repo](https://github.com/EarsEyesMouth/computerese-cross-references#h))

**Reply by wrywrywry0890** [reply]
挂接点

**Comment by FarmerChillax** [comment]
endpoint

- 端点 👀 
- 接入点 🚀
- 访问入口 🎉
- 不译 ♥

**Reply by wrywrywry0890** [reply]
端点

**Comment by ImJuanan** [comment]
**favicon**

- 站点图标 👀
- 网页图标 🚀
- favicon（不译）♥

**Reply by wrywrywry0890** [reply]
favicon
